### PR TITLE
Tickets/dm 33788

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: tests/test_socket_server.py [hunk 1]
```diff
@@ -115,6 +118,11 @@ async def test_exit(self) -> None:
         self.connected_future = asyncio.Future()
         assert self.srv.connected
         await self.assert_configure(name="TEST_EXIT")
+        # Read at least one line to make sure that the MockDevice background
+        # thread can be stopped.
+        await self.read()
+        # Make sure that all background threads of mock devices are stopped.
+        await self.write(command=common.Command.DISCONNECT, parameters={})
```
Comment (r-owen) [inline tests/test_socket_server.py:125]: Does this mean we can't just disconnect from the RPi data server? (I suppose the fact that you were already calling it at line 112 suggests the same thing).

File: tests/test_mock_command_handler.py [hunk 2]
```diff
@@ -110,6 +110,8 @@ async def test_create_device(self) -> None:
     async def test_configure(self) -> None:
         await self.command_handler.configure(configuration=self.configuration)
         assert self.configuration == self.command_handler.configuration
+        # Make sure that all background threads of mock devices are stopped.
```
Comment (r-owen) [inline tests/test_mock_command_handler.py:113]: Are you using multi-threading? If not, I suggest "tasks" instead of "threads".

**Review by r-owen — APPROVED** [review]
Looks fine. A few minor suggestions and questions (I didn't realize the RPi server had a "disconnect" command).

File: tests/test_socket_server.py (hunk 1 above)
Comment (wvreeven) [reply tests/test_socket_server.py:125]: Thanks for catching that. I added those 5 lines (and three similar ones in another test method in this class) when I was playing with multi-threading and forgot to remove them when I reverted all that. So I removed all of them now.

File: tests/test_mock_command_handler.py (hunk 2 above)
Comment (wvreeven) [reply tests/test_mock_command_handler.py:113]: This comment was removed (see previous answer to your questions here on GitHub).

**Pull request merged**